File: src/dnd/events/dicecontests/savingthrow/SacredFlame.java
package dnd.events.dicecontests.savingthrow;

import dnd.combat.DamageDiceGroup;
import dnd.data.DamageType;
import dnd.data.EventShape;
import dnd.events.Damage;
import gameobjects.entities.Entity;

public class SacredFlame extends SpellSavingThrow {

	public SacredFlame(int dcAbility) {
		super("Sacred Flame", EventShape.SINGLE_TARGET, new double[] { 60.0, 60.0 }, 0, dcAbility, Entity.DEX, 0);
	}

	@Override
	public void invokeFallout(Entity source) {
		int dieSize = 8;
		int numDice;

		if (source.getLevel() < 5) {
			numDice = 1;
		} else if (source.getLevel() < 11) {
			numDice = 2;
		} else if (source.getLevel() < 17) {
			numDice = 3;
		} else {
			numDice = 4;
		}

		Damage d = new Damage(name, this);
		d.addDamageDiceGroup(new DamageDiceGroup(numDice, dieSize, DamageType.RADIANT));
		d.invoke(source, null);
		for (Entity target : targets) {
			d.invokeClone(source, target);
		}
	}

}

File: scripts/events/contests/savingthrows/sacred_flame.lua
function define()
	Entity = luajava.bindClass("com.dndsuite.core.gameobjects.Entity")
	Event = luajava.bindClass("com.dndsuite.core.events.Event")
	
	self:setName("Sacred Flame (" .. Entity:getAbility(self:getDCAbility()) .. ")")
	self:setSaveAbility(Entity.DEX)
	self:setRange(0.0, 0.0)
	
	self:addTag(Event.SINGLE_TARGET)
	self:addTag(Event.SPELL)
	self:addTag(Event.CANTRIP)
	self:addTag("Sacred Flame")
end



function additionalEffects()
end



function additionalEffectsOnPass()
end



function damage()
	DamageType = luajava.bindClass("com.dndsuite.dnd.data.DamageType")
	
	level = source:getLevel()
	numDice = 1
	if level >= 17 then
		numDice = 4
	elseif level >= 11 then
		numDice = 3
	elseif level >= 5 then
		numDice = 2
	end
	
	damageDice = luajava.newInstance("com.dndsuite.dnd.combat.DamageDiceGroup", numDice, 8, DamageType.RADIANT)
	return damageDice
end